### PR TITLE
feat(#AVS): agentic-workflow-compatible vault structure (REQ-AVS-001–006)

### PR DESCRIPTION
**Pull request by Luis85**
$(cat <<'EOF'
## Summary

Implements the Phase 4 first target: vault structure alignment so Specorator-managed vaults are natively compatible with agentic-workflow conventions (ADR-005 / DESIGN-AVS-001).

- **REQ-AVS-001** – All feature artifacts stored under `specs/{slug}/` (configurable)
- **REQ-AVS-002** – `workflow-state.md` frontmatter aligned with agentic-workflow schema: `feature:` field, full `artifacts:` YAML block map, date-only `last_updated`, `area:` field
- **REQ-AVS-003** – Stage files named with the 12 agentic-workflow lifecycle slugs (no numeric prefixes)
- **REQ-AVS-004** – `idea.md` created at feature creation; other stage files created on stage advancement via new `AdvanceFeatureStageUseCase`
- **REQ-AVS-005** – Existing stage files preserved on advancement; user notified via `showNotice`
- **REQ-AVS-006** – Optional `area` field in feature creation; derived from slug initials when not supplied

## Changes

| File | What changed |
|---|---|
| `src/domain/feature/Feature.ts` | Add optional `area` field |
| `src/domain/feature/IFeatureRepository.ts` | Add `createStageFile()` method |
| `src/infrastructure/bridge/FeatureRepository.ts` | Fix serializer (remove `title:`, proper artifacts block, date-only `last_updated`, area derivation), fix parser to handle YAML block maps, make `save()` an upsert, add `idea.md` creation, implement `createStageFile()` |
| `src/application/feature/CreateFeatureUseCase.ts` | Add optional `area` to input |
| `src/application/feature/AdvanceFeatureStageUseCase.ts` | New use case: advance step + create stage file + save |
| `src/plugin/main.ts` | Legacy vault detection (`features/` → notice), `featuresFolder` → `specsFolder` settings migration (NFR-AVS-004) |
| `src/infrastructure/mock/fixtures.ts` | Update to new agentic-workflow schema |
| `specs/agentic-workflow-vault-structure/spec.md` | Implementation spec (new) |
| `specs/agentic-workflow-vault-structure/tasks.md` | Task breakdown with completion status (new) |

## Test plan

- [x] `npm test` — 46 tests pass (11 new tests covering REQ-AVS-001–006 acceptance criteria)
- [x] `npm run typecheck` — zero errors
- [x] `npm run lint` — zero errors
- [ ] Manual: `npm run dev` — verify features listed under `specs/` with correct frontmatter in browser mode

## Related

- ADR-005: `docs/adr/ADR-005-agentic-workflow-vault-structure.md`
- Design: `specs/agentic-workflow-vault-structure/design.md`
- Requirements: `specs/agentic-workflow-vault-structure/requirements.md`
- Roadmap: #47 (Phase 4 first target)

https://claude.ai/code/session_013KA2BcanTNSCki7TZyBgtt
EOF
)

---
_Generated by [Claude Code](https://claude.ai/code/session_013KA2BcanTNSCki7TZyBgtt)_

### CODE REVIEW

File: src/application/feature/AdvanceFeatureStageUseCase.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    if (!feature) {
+      return err(new Error(`Feature "${input.featureId}" not found`))
+    }
+
+    const advancedResult = feature.advanceStep()
+    if (!advancedResult.ok) return advancedResult
+
+    const advanced = advancedResult.value
+
+    const fileResult = await this.repository.createStageFile(advanced, advanced.currentStep)
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/application/feature/AdvanceFeatureStageUseCase.ts]: Fixed in ede3910. `createStageFile` is now skipped when `advanced.isComplete` is true, so advancing from step 12 → 13 (the completion sentinel) saves the updated `workflow-state.md` without attempting to create a step-13 file. Added a regression test covering this path.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 2]
```diff
@@ -2,45 +2,90 @@ import { ok, err, type Result } from '@/domain/shared/Result'
...
+		} else {
+			status = 'pending'
+		}
+		return `  ${slug}: ${status}`
+	}).join('\n')
+}
+
 function serializeFeature(feature: Feature): string {
 	const p = feature.toPlainObject()
 	const currentStage = FEATURE_STEPS[p.currentStep - 1] ?? 'idea'
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in 3767790. The stage index is now clamped to `FEATURE_STEPS.length - 1` so any `currentStep > 12` resolves to `retrospective` rather than falling through to the old `?? 'idea'` sentinel. The completion regression test now also asserts `current_stage: retrospective`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 3]
```diff
@@ -109,18 +180,47 @@ export class FeatureRepository implements IFeatureRepository {
...
 		try {
 			const folder = `${this.settings.specsFolder}/${feature.slug.toString()}`
 			await this.bridge.createFolder(folder)
 			const path = this.metaPath(feature.slug.toString())
+			const isNew = !(await this.bridge.fileExists(path))
+			await this.bridge.writeFile(path, serializeFeature(feature))
+			if (isNew) {
+				const date = feature.createdAt.toISOString().slice(0, 10)
+				const ideaPath = this.stagePath(feature.slug.toString(), 'idea')
+				await this.bridge.writeFile(ideaPath, buildStageStub('idea', feature.slug.toString(), feature.title, date))
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in cf9f9d8. `save()` now checks whether `idea.md` already exists before writing it — if it does, a notice is shown and the file is preserved, consistent with the behaviour in `createStageFile()`. Added a regression test for the preserved-idea.md path.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 4]
```diff
@@ -2,45 +2,93 @@ import { ok, err, type Result } from '@/domain/shared/Result'
...
+	}).join('\n')
+}
+
 function serializeFeature(feature: Feature): string {
 	const p = feature.toPlainObject()
-	const currentStage = FEATURE_STEPS[p.currentStep - 1] ?? 'idea'
+	// Clamp to the last valid index so completed features (currentStep > 12)
+	// write `retrospective` rather than falling back to `idea`.
+	const stageIndex = Math.min(p.currentStep - 1, FEATURE_STEPS.length - 1)
+	const currentStage = FEATURE_STEPS[stageIndex]
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in 753b4a4. Added `Math.max(0, ...)` as the lower bound so `currentStep ≤ 0` maps to index 0 (`idea`) rather than `-1` (`undefined`). Added a regression test using `currentStep: 0` to guard this path.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 5]
```diff
@@ -2,45 +2,93 @@ import { ok, err, type Result } from '@/domain/shared/Result'
...
+	const area = p.area || deriveArea(p.slug)
+	const lastUpdated = p.updatedAt.toISOString().slice(0, 10)
 	return [
 		'---',
 		`id: ${p.id}`,
 		`slug: ${p.slug}`,
 		`feature: "${p.title.replace(/"/g, '\\"')}"`,
-		`title: "${p.title.replace(/"/g, '\\"')}"`,
-		`area: ""`,
+		`area: ${area}`,
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in 753b4a4. The serializer now strips all non-uppercase-letter characters from `area` before writing (`/[^A-Z]/g`), so inputs like `R&D` or `TEAM:EU` are sanitized to `RD` / `TEAMEU` and the scalar is always a plain safe YAML string. Falls back to slug-derived initials if sanitization empties the value. Added a test for the `R&D` → `RD` case.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 6]
```diff
@@ -109,18 +181,53 @@ export class FeatureRepository implements IFeatureRepository {
...
+	 * Upsert: write workflow-state.md for a new or updated feature.
+	 * On first creation (file did not exist), also writes the idea.md stub.
+	 */
 	async save(feature: Feature): Promise<Result<void>> {
 		try {
 			const folder = `${this.settings.specsFolder}/${feature.slug.toString()}`
 			await this.bridge.createFolder(folder)
 			const path = this.metaPath(feature.slug.toString())
+			const isNew = !(await this.bridge.fileExists(path))
+			await this.bridge.writeFile(path, serializeFeature(feature))
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in d62e9d7. `findBySlug` now throws when the file exists but `deserializeFeature` returns `null`, so the create path can never silently overwrite it. `CreateFeatureUseCase` wraps the call in a try/catch and surfaces the error. Added a test that pre-seeds a malformed `workflow-state.md` and asserts that the file is untouched after a failed create attempt.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 7]
```diff
@@ -2,45 +2,94 @@ import { ok, err, type Result } from '@/domain/shared/Result'
...
+
 function serializeFeature(feature: Feature): string {
 	const p = feature.toPlainObject()
-	const currentStage = FEATURE_STEPS[p.currentStep - 1] ?? 'idea'
+	// Clamp index to [0, last] so currentStep ≤ 0 or > 12 both produce a valid stage.
+	const stageIndex = Math.max(0, Math.min(p.currentStep - 1, FEATURE_STEPS.length - 1))
+	const currentStage = FEATURE_STEPS[stageIndex]
+	// Strip non-uppercase-letter characters so the area scalar is always safe YAML.
+	const rawArea = p.area || deriveArea(p.slug)
+	const area = rawArea.replace(/[^A-Z]/g, '').slice(0, 5) || deriveArea(p.slug)
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in d62e9d7. Sanitization is now applied as a shared `sanitize` helper used on both the user-supplied value and the `deriveArea` fallback. The final fallback is the hardcoded `'XX'` so the value is never empty. Area is also now written **quoted** (`area: "DM"`) so YAML parsers always see a string scalar, even if a future edge case slips through the letter-only filter. Tests updated to assert the quoted format; added a numeric-title test to cover the `area: \d` regression.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 8]
```diff
@@ -101,26 +174,66 @@ export class FeatureRepository implements IFeatureRepository {
...
 				this.bridge.showNotice(
-					`Spec "${feature.slug.toString()}" already exists — skipping overwrite.`,
+					`Specorator: ${meta.slug}.md already exists — keeping your version.`,
 				)
-				return err(new Error(`Spec "${feature.slug.toString()}" already exists`))
+				return ok(undefined)
 			}
-			await this.bridge.writeFile(path, serializeFeature(feature))
+			const date = new Date().toISOString().slice(0, 10)
+			await this.bridge.writeFile(path, buildStageStub(meta.slug, feature.slug.toString(), feature.title, date))
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts:255]: Fixed in 24e8beb. `delete()` now calls `listFiles(folder)` and deletes every file in the feature folder in parallel, so `idea.md`, stage artifacts, and `workflow-state.md` are all removed together. An empty folder stub may remain (Obsidian hides these). Added a test that advances a feature to step 2 then deletes it, asserting all three files are gone.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/application/feature/AdvanceFeatureStageUseCase.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    const advanced = advancedResult.value
+
+    // When currentStep exceeds the last stage (feature complete), there is no
+    // stage file to create — skip the file step and persist the completion state.
+    if (!advanced.isComplete) {
+      const fileResult = await this.repository.createStageFile(advanced, advanced.currentStep)
+      if (!fileResult.ok) return fileResult
+    }
+
+    const saveResult = await this.repository.save(advanced)
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/application/feature/AdvanceFeatureStageUseCase.ts:40]: Fixed in 0cba6c5. `save()` is now called before `createStageFile()`, so if the persist step fails no stage artifact is written and the vault stays consistent. On a successful save, writing the stage file is best-effort and any failure is still surfaced to the caller.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 10]
```diff
@@ -2,45 +2,95 @@ import { ok, err, type Result } from '@/domain/shared/Result'
...
+ */
 function parseFrontmatter(content: string): Record<string, string> {
 	const match = /^---\n([\s\S]*?)\n---/.exec(content)
 	if (!match) return {}
 
 	return Object.fromEntries(
 		match[1].split('\n').flatMap((line) => {
+			// Skip indented lines (block map child values)
+			if (line.startsWith(' ') || line.startsWith('\t')) return []
 			const colonIdx = line.indexOf(':')
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in 0cba6c5. The parser now tracks an `inArtifactsBlock` flag that is set only when the `artifacts:` key is encountered and cleared as soon as a non-indented line follows. Indented lines are skipped only while inside that block, so any top-level key that a formatter chooses to indent is parsed normally.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/application/feature/AdvanceFeatureStageUseCase.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    // Persist workflow-state first so the step counter is durable before the
+    // stage artifact file is written.  If save fails we leave no orphaned file.
+    const saveResult = await this.repository.save(advanced)
+    if (!saveResult.ok) return saveResult
+
+    // When currentStep exceeds the last stage (feature complete), there is no
+    // stage file to create — skip the file step.
+    if (!advanced.isComplete) {
+      const fileResult = await this.repository.createStageFile(advanced, advanced.currentStep)
+      if (!fileResult.ok) return fileResult
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/application/feature/AdvanceFeatureStageUseCase.ts]: Fixed in 77c8f6a. Reverted to file-first order with an explicit comment explaining the idempotency argument: `createStageFile` returns `ok` without overwriting when the file already exists, so a retry after a failed `save` will find the artifact present, skip the write, and re-attempt the save at the correct `currentStep`. The save-first order is genuinely unsafe for exactly the reason described — a `createStageFile` failure after a successful save leaves `currentStep` already incremented, causing double-advance on retry.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 12]
```diff
@@ -101,26 +182,66 @@ export class FeatureRepository implements IFeatureRepository {
...
+	 */
 	async save(feature: Feature): Promise<Result<void>> {
 		try {
 			const folder = `${this.settings.specsFolder}/${feature.slug.toString()}`
 			await this.bridge.createFolder(folder)
 			const path = this.metaPath(feature.slug.toString())
+			const isNew = !(await this.bridge.fileExists(path))
+			await this.bridge.writeFile(path, serializeFeature(feature))
+			if (isNew) {
+				const ideaPath = this.stagePath(feature.slug.toString(), 'idea')
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in b59f5ed. `idea.md` is now written before `workflow-state.md`. Since `idea.md` creation is idempotent (preserves an existing file and returns `ok`), a failed `workflow-state.md` write leaves the folder without a metadata file — `findBySlug` returns `null` and `CreateFeatureUseCase` can retry cleanly. The old order was unsafe: a successful `workflow-state.md` write followed by a failed `idea.md` write left a valid metadata file that blocked any retry via the duplicate-slug check.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/application/feature/CreateFeatureUseCase.ts [hunk 13]
```diff
@@ -6,6 +6,8 @@ import type { UseCase } from '../shared/UseCase'
 
 export interface CreateFeatureInput {
   readonly title: string
+  /** Optional 3–5 uppercase-letter area code (e.g. "DM" for dark-mode). Derived from slug if omitted. */
+  readonly area?: string
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/application/feature/CreateFeatureUseCase.ts:10]: Fixed in 16dc136. The optional `area` input is now wired end-to-end:

- `CreateFeatureForm.vue` emits a `{ title, area? }` payload, with a new optional area input (`maxlength=5`).
- `useFeatures.createFeature(title, area?)` forwards the value to `CreateFeatureUseCase`.
- `HomeView` and `FeaturesView` `handleCreate` reshaped to accept the payload.
- Added `feature.area` / `feature.areaPlaceholder` in en + de locales.

When the field is left blank, the serializer still derives the area from the slug initials (existing fallback). REQ-AVS-006 is now reachable from the plugin UI. Typecheck clean, 53 tests pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/infrastructure/bridge/FeatureRepository.ts [hunk 14]
```diff
@@ -2,45 +2,103 @@ import { ok, err, type Result } from '@/domain/shared/Result'
...
+ */
+function buildArtifactsBlock(currentStep: number): string {
+	return FEATURE_STEPS.map((slug, idx) => {
+		const stepNum = idx + 1
+		let status: string
+		if (stepNum < currentStep) {
+			status = 'complete'
+		} else if (stepNum === currentStep) {
+			status = currentStep === 1 ? 'complete' : 'in-progress'
+		} else {
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/infrastructure/bridge/FeatureRepository.ts]: Fixed in 4dae400. `buildArtifactsBlock` now clamps the input to `Math.max(1, currentStep)` before computing statuses, matching the lower-bound clamp already applied to `stageIndex` in `serializeFeature`. `currentStep ≤ 0` now produces `idea: complete` (consistent with `current_stage: idea`) rather than `idea: pending`. Added a regression assertion to the existing `currentStep: 0` test.

---
_Generated by [Claude Code](https://claude.ai/code)_
